docs/source/json-api/index.rst

```diff
@@ -32,140 +32,123 @@ We welcome feedback about the JSON API on `our issue tracker
-    $ daml new iou-quickstart-java --template quickstart-java
-    $ cd iou-quickstart-java/
-    $ daml build
-    $ daml sandbox --wall-clock-time --ledgerid MyLedger ./.daml/dist/quickstart-0.0.1.dar
-    $ daml json-api --ledger-host localhost --ledger-port 6865 --http-port 7575
+The JSON API essentially performs two separate tasks:
 
-Choosing a party
-****************
+1. It talks to the Ledger API to get data it needs to operate, for this it needs to may need to provide an *authentication token* to the Ledger API if the Ledger API requires *authentication*.
+2. It handles the requests coming from one or more Parties, for this each party needs to provide an *authorization token* with each request it sends to the JSON API.
 
-Every request requires you to specify a party and some other settings,
-with a JWT token.  Normal HTTP requests pass the token in an
-``Authentication`` header, while WebSocket requests pass the token in a
-subprotocol.
+Authentication
+--------------
 
-In testing environments, you can use https://jwt.io to generate your
+The authentication token is used exclusively for maintaining the internal list of known packages and templates.
+
+.. note:: At no point should an authentication token be provided to an end user, these are for internal use only.
+
+Every authentication token is different and will depend on what your specific ledger or ledger operator requires.
+The JSON API server requires no access to party-specific data, only access to the ledger identity and package services.
+A token issued for the HTTP JSON API server should contain enough claims to contact these two services but no more than that.
+Please refer to your ledger operator's documentation to find out how.
+
+Once you have retrieved your access token, you can provide it to the JSON API by storing it in a file
+and starting ``daml json-api`` with the flag ``--access-token-file /path/to/your/token.file``.
+
+If the token cannot be read from the provided path or the Ledger API reports an authentication error
+(for example due to token expiration), the JSON API will report the error via logging. 
+
+.. note:: If the token file is updated with a new token it will be picked up at the next attempt to send a request. You can use this to handle cases where an old token expires without interrupting your JSON API service.
+
+Authorization
+-------------
+
+Every request from a client to the JSON API requires you to specify a party and some other settings,
+with a JWT token.  Normal HTTP requests pass the token in an ``Authorization`` header, 
+while WebSocket requests pass the token in a subprotocol.
+
+.. note:: While the JSON API receives the token it doesn't handle it itself. Upon receiving a token it will pass it on to the underlying Ledger's AuthService which will then determine if the token is valid and authorized. As such each token format is ledger/implementation-specific and the below example reflects only the format that is used for the DAML Sandbox.
```

Comment:
> As such each token format is ledger/implementation-specific and the below example reflects only the format that is used for the DAML Sandbox.

This is at least partially wrong. The JSON API doesn’t validate the token but it does decode it and extract the claims from it. That’s why you don’t have to specify the party, application id or ledger id when submitting a command via the JSON API. So while there can be additional fields that will be validated by the ledger, you need to have the standard format. In fact there are even more limitations: The JSON API requires `applicationId`, `ledgerId` and requires that the set of ``act_as `union` readAs`` contains exactly one party.

Reply:
Where does `readAs` come from?